union-ai-admin/aws/script/generate.py

```diff
@@ -563,84 +673,60 @@ def create_provisioner_policy(role_type):
                     ],
                     Resource=["*"],
                 ),
                 Statement(
                     Effect=Allow,
                     Action=[
-                        Action("ec2", "CreateVpcEndpoint"),
-                        Action("ec2", "DeleteVpcEndpoints"),
-                        Action("ec2", "CreateTags"),
-                    ],
-                    Resource=[
-                        Sub("arn:aws:ec2:${AWS::Region}:${AWS::AccountId}:vpc/vpc*"),
-                        Sub(
-                            "arn:aws:ec2:${AWS::Region}:${AWS::AccountId}:vpc-endpoint/*"
-                        ),
-                        Sub(
-                            "arn:aws:ec2:${AWS::Region}:${AWS::AccountId}:route-table/*"
-                        ),
-                        Sub("arn:aws:ec2:${AWS::Region}:${AWS::AccountId}:subnet/*"),
+                        Action("autoscaling", "CreateAutoScalingGroup"),
+                        Action("autoscaling", "DeleteAutoScalingGroup"),
+                        Action("autoscaling", "SetInstanceProtection"),
                     ],
+                    Resource=["*"],
+                    Condition=Condition(
+                        StringEqualsIfExists(
+                            "aws:RequestTag/ManagedByUnion",
+                            "true",
+                        )
+                    ),
                 ),
             ],
         ),
     )
 
 
-def create_admin_policy(role_type):
+def create_terraform_policy(role_type):
     """
     Create a managed policy for the admin IAM role used by Terraform.
     :return: The ManagedPolicy object.
     """
     return ManagedPolicy(
-        "UnionaiAdminPermission",
-        ManagedPolicyName=f"admin-policy-for-{role_type}-role",
+        "UnionaiTerraformPermission",
+        ManagedPolicyName=f"terraform-policy-for-{role_type}-role",
         PolicyDocument=PolicyDocument(
             Version="2012-10-17",
             Statement=[
-                Statement(
-                    Effect=Allow,
-                    Action=[
-                        Action("eks", "CreateNodegroup"),
-                        Action("eks", "TagResource"),
-                        Action("eks", "UntagResource"),
-                        Action("eks", "DeleteNodegroup"),
-                    ],
-                    Resource=[
-                        Sub(
-                            "arn:aws:eks:${AWS::Region}:${AWS::AccountId}:cluster/opta-*"
-                        ),
-                        Sub(
-                            "arn:aws:eks:${AWS::Region}:${AWS::AccountId}:nodegroup/opta-*/opta-*/*"
-                        ),
-                        Sub(
-                            "arn:aws:eks:${AWS::Region}:${AWS::AccountId}:nodegroup/opta-*"
-                        ),
-                    ],
-                ),
                 Statement(
                     Effect=Allow,
                     Action=[
```

Comment:
We should not use `Action("s3", "*")`. We should be able to define which s3: actions we need to perform.